policy/modules/system/fstools.te

```diff
@@ -113,16 +113,23 @@ files_getattr_boot_dirs(fsadm_t)
 files_list_home(fsadm_t)
 files_read_usr_files(fsadm_t)
 files_read_etc_files(fsadm_t)
+files_read_fstab(fsadm_t)
 files_manage_lost_found(fsadm_t)
 # Write to /etc/mtab.
 files_manage_etc_runtime_files(fsadm_t)
 files_etc_filetrans_etc_runtime(fsadm_t, file)
 
+fs_ioctl_cgroup_dirs(fsadm_t)
 fs_rw_all_image_files(fsadm_t)
 fs_search_auto_mountpoints(fsadm_t)
 fs_getattr_xattr_fs(fsadm_t)
 fs_rw_ramfs_pipes(fsadm_t)
 fs_rw_tmpfs_files(fsadm_t)
+# for zfs mount/unmount commands
+fs_mount_xattr_fs(fsadm_t)
+fs_unmount_xattr_fs(fsadm_t)
+# for zfs filesystem rootcontext property
+fs_relabelfrom_xattr_fs(fsadm_t)
```

Comment:
I think you're right. I wanted to avoid this as much as possible but it seems like the tools have enough needed accesses that creeps into the other fs-related domains that it's probably best to split it out.

I'll drop the changes specifically related to these tools from this series and work on that. It will probably end up as a new module.